spec/indexers/app_indexer_spec.rb

```diff
@@ -24,4 +24,34 @@
       expect(solr_document.fetch("account_cname_tesim")).to eq(account.cname)
     end
   end
+
+  describe "#generate_solr_document" do
+    context "when given a date with a YYYY-MM-DD format" do
```

Comment:
What about "2024-1-12"? Or "2024-13-50" or "2024-12-24-0"?

I think we should coerce "2024-1-12" to "2024-01-12" (using `sprintf` perhaps)?

Reply:
Hmmm, I suppose we could, but the help text clearly says what date formats are allowed
https://github.com/scientist-softserv/palni-palci/blob/i973-date-sorting/config/locales/hyrax.en.yml#L1142